components/sekhras/actions/sekhraEstimation.js

```diff
@@ -0,0 +1,70 @@
+const getRoute = require('../../../web/api/mapQuest/route');
+const logger = require('../../../config/logger');
+const HTTP = require('../../../constants/statusCode');
+const statusTypes = require('../constants/status');
+const Sekhra = require('../sekhra');
+const Customer = require('../../user/customer');
+const Coursier = require('../../user/coursier');
+const userTypes = require('../../user/constants/types');
+const getClosest = require('../helpers/closestCoursier');
+const formatShapePoints = require('../helpers/formatShapePoints');
+const Notification = require('../../notification/notif');
+
+module.exports = estimateSekhra = async (sekhra, io) => {
+    const { from, to } = sekhra;
+    const [fromLat, fromLong] = from;
+    let coursiers;
+    try {
+        // Find the 5 closest actif coursiers
+        coursiers = await findClosestActifCoursier({
+            actif: "true",
+            location: {
+                $near: {
+                    $geometry: {
+                        type: 'Point',
+                        coordinates: [fromLat, fromLong]
+                    }
+                }
+            }
+        });
+    } catch (error) {
+        logger.error(error);
+        return res.status(HTTP.SERVER_ERROR).send({ error });
+    }
+    let users = [];
+    for (let i = 0; i < coursiers.length; i++) {
+        if (coursiers[i].currentSekhras.length === 0) {
+            users.push(coursiers[i]._id);
+        } else {
+            for (let i = 0; i < coursiers[i].currentSekhras.length; i++) {
+                const sekhra = coursiers[i].currentSekhras[i];
+                try {
+                    let coursierSekhra = await Sekhra.findById(
+                        sekhra._id
+                    );
```

Comment:
You're calling findById for each iteration.

Reply:
The value of sekhra change in each iteration.

Reply:
Could you explain how this function works? why would the value change with each iteration ?

Reply:
A Rider can be shared by multiple customers, they all need to be notified to accept or reject sharing the sekhra, so we go through all the sekhras that the rider have at the moment and retrieve the customer of each sekhra (sorry for answering late).